### PR TITLE
Update to go 1.20.5

### PR DESCRIPTION
**Pull request by jrouzierinverse**
# Description
Update to go 1.20.5

# Delete branch after merge
YES

# Checklist
(REQUIRED) - [yes, no or n/a]
- [ ] Document the feature
- [ ] Add unit tests
- [ ] Add acceptance tests (TestLink)

# NEWS file entries
## Enhancements
* Update to go 1.20.5

### CODE REVIEW

File: go/go.sum [hunk 1]
```diff
@@ -369,7 +364,6 @@ github.com/hashicorp/serf v0.8.2/go.mod h1:6hOLApaqBFA1NXqRQAsxw9QxuDEvNxSQRwA/J
 github.com/hpcloud/tail v1.0.0 h1:nfCOvKYfkgYP8hkirhJocXT2+zOD8yUNjXaWfTlyFKI=
 github.com/hpcloud/tail v1.0.0/go.mod h1:ab1qPbhIpdTxEkNHXyeSf5vhxWSCs/tWer42PpOxQnU=
 github.com/hudl/fargo v1.3.0/go.mod h1:y3CKSmjA+wD2gak7sUSXTAoopbhU08POFhmITJgmKTg=
-github.com/ianlancetaylor/demangle v0.0.0-20200824232613-28f6c0f3b639 h1:mV02weKRL81bEnm8A0HT1/CAelMQDBuQIfLw8n+d6xI=
 github.com/ianlancetaylor/demangle v0.0.0-20200824232613-28f6c0f3b639/go.mod h1:aSSvb/t6k1mPoxDqO4vJh6VOCGPwU4O0C2/Eqndh1Sc=
```
Comment (JeGoi) [inline go/go.sum:367]: Why do you keep the go.mod for this one here when github.com/google/pprof looks completely removed?

**Review by JeGoi — COMMENTED** [review]
Just want to be sure that other go.mod not deleted are usefull here

**Review by JeGoi — APPROVED** [review]

File: go/go.sum (hunk 1 above)
Comment (JeGoi) [reply go/go.sum:367]: Well, still have that question in fact...

Comment (jrouzierinverse) [reply go/go.sum:367]: It could be used for a different module, or it is a dependency of a dependency

**Comment by extrafu** [comment]
@nqb Make sure to adjust the CI/CD after this gets merged.

**Comment by extrafu** [comment]
@JeGoi and @jrouzierinverse are we OK to merge that ASAP? We don't want to wait last minute and we need to let @nqb know about this change.

**Review by JeGoi — APPROVED** [review]
Same as before. Agree

**Comment by nqb** [comment]
Images have been updated with new Golang version.

**Pull request merged**